src/mds/MDSContext.cc

```diff
@@ -95,6 +95,7 @@ void MDSIOContextBase::complete(int r) {
   if (mds->is_daemon_stopping()) {
     dout(4) << "MDSIOContextBase::complete: dropping for stopping "
             << typeid(*this).name() << dendl;
+    delete this;
```

Comment:
It's preferred to call `Context::complete` instead (see elsewhere in this file).

Reply:
@batrick Thanks for your review:-)

It seems Context::complete would further call the "finish" method. I thought it may cause other problem, since it's shutting down. Is that possible? Thanks:-)

Reply:
I think normally a subclass of Context would only define `::finish` so this is all a little backwards. Maybe Zheng can chime in on why he did it this way.

Reply:
@batrick I've just modified the code, please take a look:-)

Reply:
@ukernel